rs/bitcoin/kyt/src/fetch.rs

```diff
@@ -74,13 +77,24 @@ pub trait FetchEnv {
         &self,
         txid: Txid,
     ) -> TryFetchResult<impl futures::Future<Output = Result<FetchResult, Infallible>>> {
-        let max_response_bytes = match state::get_fetch_status(txid) {
-            None => INITIAL_MAX_RESPONSE_BYTES,
+        let default_provider = Provider::new_with_default(self.btc_network());
```

Comment:
Maybe for a follow-up PR: this has the drawback that the same provider will be tried first, which is a bit sub-optional if a provider is offline for some time period. We could maybe optimize this by saving the provider used in the last successful request. WDYT?

Reply:
A more general question would be, why would we have a default go-to provider? Why not always use round-robin or a random provider to flatten the load?

Reply:
I think we have to make sure that all of the providers are used regularly. Otherwise, if only one is used, we may not notice that the others don't work anymore for whatever reason and then we falsely believe that there are backups.